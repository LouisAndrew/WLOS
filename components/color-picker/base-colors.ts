// tailwind colors on (500)
export const colors = [
  'red',
  'yellow',
  'green',
  'blue',
  'indigo',
  'pink',
  'black',
  'white',
] as const
export type Color = typeof colors[number]
export type ColorCode = Record<Color, string>
export const colorCodes: ColorCode = {
  red: '#EF4444',
  yellow: '#F59E0B',
  green: '#10B981',
  blue: '#3B82F6',
  indigo: '#6366F1',
  pink: '#EC4899',
  black: '#0E0E0E',
  // black: '#8A8A8A',
  white: '#FFFFFF',
}

export const colorTable: [Color, string][] = Object.keys(colorCodes).map((key) => [
  key as Color,
  colorCodes[key],
])
