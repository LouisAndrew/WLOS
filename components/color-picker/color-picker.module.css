.container {
  @apply px-5 py-4;
}

.color-selector__wrapper {
  @apply flex flex-wrap mt-2;
}

.color-selector {
  @apply h-5 w-5 flex flex-center rounded-full ml-0 mr-4 my-1 focus:outline-none flex-shrink-0;
}

.color-selector__marker {
  /* background-color: #333; */
  @apply duration-200 h-2 w-2 opacity-0 group-hover:opacity-100 rounded-full bg-bg-gray;
}

.color-selector__marker.is-active {
  @apply w-3 h-3 opacity-100;
}

.custom-color__preview {
  @apply h-5 w-5 border-gray-500 border-2 rounded-full duration-200 mr-3 flex-shrink-0;
}

.custom-color__input-wrapper {
  @apply bg-primary-gray mt-2 rounded flex items-center py-1 px-3 w-full border-2 border-bg-gray;
}

.custom-color__input {
  @apply font-display text-lg placeholder-gray-500 uppercase block w-full;
}
