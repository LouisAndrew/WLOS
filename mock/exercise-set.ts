import { ExerciseSet } from '@t/ExerciseSet'
import { Metric } from '@t/Metric'
import { Review } from '@t/Review'

const mockExerciseSet: ExerciseSet = {
  setNumber: 1,
  repsCount: 12,
  weightValue: 20,
  weightMetric: Metric.KG,
  review: Review.UP,
}

const mockExerciseSetWithoutReview: ExerciseSet = {
  ...mockExerciseSet,
  review: undefined,
}

export { mockExerciseSet, mockExerciseSetWithoutReview }
