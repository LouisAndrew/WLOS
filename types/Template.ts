import { DocumentReference } from '@lib/API/firebase'
import { ExerciseDBSchema } from './Exercise'
import { UserDBSChema } from './User'

/**
 * type of the template
 * @value 0 for workout template
 * @value 1 for warmup template
 */
export type TemplateType = 0 | 1

type TemplateBase = {
  /**
   * type of the template
   * @value 0 for workout template
   * @value 1 for warmup template
   */
  type: TemplateType
  /**
   * tag ids of the template
   * ! WIP
   */
  tags: string[] // todo
  /**
   * name of the template
   */
  name: string
  /**
   * color code of the template.
   * TODO: decide if this should be editable
   */
  color: string
  createdBy: any
  exercises: any
}

export type TemplateDBSchema = TemplateBase & {
  templateId: string
  /**
   * reference to user
   */
  createdBy: DocumentReference<UserDBSChema>
  /**
   * DB schema for exercises
   */
  exercises: DocumentReference<ExerciseDBSchema>[]
}

export type Template = TemplateBase & {
  /**
   * id of the user who created the template
   */
  createdBy: string
  /**
   * exercises type that has been converted
   */
  exercises: TemplateExercise[]
}

export type TemplateExercise = {
  /**
   * data of the exercise
   * @see ExerciseTable
   */
  exerciseData: { name: string; exerciseId: string }
  /**
   * nums of sets of the exercise in `Range` type
   * @see Range
   */
  sets: string
  /**
   * nums of reps of the exercise in `Range` type
   * @see Range
   */
  reps: string
  /**
   * order of the exercise within the template
   */
  order: number
}
