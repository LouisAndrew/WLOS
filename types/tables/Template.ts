import { UserDataTable } from './UserData'
import { Table } from './Table'

/**
 * type of the template
 * @value 0 for workout template
 * @value 1 for warmup template
 */
export type TemplateType = 0 | 1

export type TemplateTable = Table & {
  /**
   * type of the template
   * @value 0 for workout template
   * @value 1 for warmup template
   */
  type: TemplateType
  /**
   * user who created the template
   */
  created_by: number | UserDataTable
  /**
   * tag ids of the template
   * ! WIP
   */
  tags?: number[]
  /**
   * name of the template
   */
  name: string
  /**
   * color code of the template.
   * todo: decide if this should be editable
   */
  color: string
}
