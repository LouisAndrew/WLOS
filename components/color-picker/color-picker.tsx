import React, { FC, useEffect, useState } from 'react'
import classname from 'classnames/bind'
import { colorTable } from './base-colors'

import styles from './color-picker.module.css'
const cx = classname.bind(styles)

export type Props = {
  /**
   * default selected color
   */
  defaultSelected?: string
  /**
   * function to handle when a color button is clicked or when the custom color input is filled
   */
  onColorChange: (colorCode: string) => void
}

const ColorPicker: FC<Props> = ({ defaultSelected, onColorChange }) => {
  const checkIfColorCustomed = (color: string) =>
    colorTable.findIndex(([_, colorCode]) => colorCode === color) === -1

  const [selectedColor, setSelectedColor] = useState(defaultSelected || '')
  const [isColorCustomed, setIsColorCustomed] = useState(checkIfColorCustomed(selectedColor))

  const handleChangeCustomColor = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value

    if (value.length > 7) {
      return
    }

    if (value === '' || value[0] === '#') {
      setSelectedColor(value)
      return
    }

    setSelectedColor(`#${value}`)
  }

  useEffect(() => {
    setIsColorCustomed(checkIfColorCustomed(selectedColor))
    onColorChange(selectedColor)
  }, [selectedColor])

  const customColorPreviewClass = cx({
    'custom-color__preview': true,
    'is-customed': isColorCustomed,
  })

  return (
    <div className={styles.container} data-testid="wrapper">
      <h3 className="heading heading-3 pb-2">COLORS</h3>
      <div className={styles['color-selector__wrapper']}>
        {colorTable.map(([colorName, colorCode]) => {
          const isColorActive = selectedColor === colorCode
          const colorId = `color-selector-${colorName}`
          const colorSelectorClass = cx({
            'color-selector': true,
            group: true,
            'is-active': isColorActive,
          })
          const colorSelectorMarkerClass = cx({
            'color-selector__marker': true,
            'is-active': isColorActive,
          })

          return (
            <button
              data-testid={colorId}
              key={colorId}
              style={{ backgroundColor: colorCode }}
              className={colorSelectorClass}
              onClick={() => setSelectedColor(colorCode)}
            >
              <div className={colorSelectorMarkerClass} />
            </button>
          )
        })}
      </div>
      <label htmlFor="custom-color" className="block body mt-5">
        Custom color
        <div
          className={styles['custom-color__input-wrapper']}
          style={{ borderColor: selectedColor }}
        >
          <div
            data-testid="custom-color-preview"
            className={customColorPreviewClass}
            style={{ backgroundColor: selectedColor, borderColor: selectedColor }}
          />
          <input
            type="text"
            id="custom-color"
            onChange={handleChangeCustomColor}
            placeholder="#"
            value={selectedColor}
            className={styles['custom-color__input']}
          />
        </div>
      </label>
    </div>
  )
}

export default ColorPicker
