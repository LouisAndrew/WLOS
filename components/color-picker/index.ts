import ColorPicker from './color-picker'
import { colorCodes, colors, colorTable } from './base-colors'

export { ColorPicker, colorCodes, colors, colorTable }
