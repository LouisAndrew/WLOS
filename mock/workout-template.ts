import { TemplateTableWithData } from '@t/tables/Template'
import { colorCodes } from '@c/color-picker'
import { mockExerciseTable, mockExerciseTable2 } from './exercise'
import { mockExerciseSetWithoutReview } from './exercise-set'

export const defaultTemplateTableWithData: TemplateTableWithData = {
  id: -1,
  type: 0,
  created_by: -1,
  name: '',
  color: colorCodes.black,
  exercises: [],
}

export const filledTemplateTableWithData: TemplateTableWithData = {
  id: 12,
  type: 0,
  created_by: -1,
  name: 'PRESET WORKOUT',
  color: colorCodes.red,
  exercises: [
    {
      exerciseData: mockExerciseTable,
      sets: '3-4',
      reps: '12',
      order: 1,
    },
    {
      exerciseData: mockExerciseTable2,
      sets: '2-3',
      reps: '19',
      order: 0,
    },
  ],
}
