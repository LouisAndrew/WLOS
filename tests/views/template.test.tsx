import { defaultTemplateTableWithData, filledTemplateTableWithData } from '@/mock/workout-template'
import { colorCodes } from '@c/color-picker'
import { render, cleanup, fireEvent } from '@testing-library/react'
import { changeEvent } from '@tests/utils/changeEvent'
import Template, { PageState, Props } from '@v/template/template'

afterEach(cleanup)

const component = (args: Partial<Props> = {}) =>
  render(
    <Template defaultState={PageState.EDIT} template={defaultTemplateTableWithData} {...args} />
  )

describe('Template page', () => {
  it('renders the component', () => {
    expect(component().getByTestId('template-wrapper')).toBeInTheDocument()
  })

  it('matches the snapshot', () => {
    expect(component()).toMatchSnapshot()
  })

  it('renders the color picker and change the template color based on the color user picks', () => {
    const { getByText: txt, getByTestId: tid, getByLabelText: l } = component({})
    fireEvent.click(txt('CHANGE COLOR'))
    expect(tid('color-picker-wrapper')).toBeInTheDocument()

    const value = colorCodes.indigo
    fireEvent.click(l(value))
    expect(l('Template Color')).toHaveStyle(`background-color: ${value}`)
  })

  it('renders the save button if any changes is made on the component', () => {
    const { getAllByLabelText: altxt, getByTestId: t, queryByTestId: qt } = component({
      template: filledTemplateTableWithData,
    })
    const exerciseNameInputs = altxt('Exercise Name')
    expect(qt('save-btn')).not.toBeInTheDocument() // should not be displayed at first

    fireEvent.change(exerciseNameInputs[0], changeEvent('Random name'))

    expect(t('save-btn')).toBeInTheDocument()
  })

  it('allows user to change the page state', () => {
    const { queryAllByTestId: qat, getByTestId: t } = component()
    fireEvent.click(t('change-state-btn'))
    expect(qat('state-btn').length).toBe(Object.keys(PageState).length - 2) // minus current state and "creating" state
  })

  it('displays a modal when any changes are not saved', () => {
    const {
      getAllByLabelText: altxt,
      getByTestId: t,
      queryAllByTestId: qat,
      queryByTestId: qt,
    } = component({
      template: filledTemplateTableWithData,
    })
    const exerciseNameInputs = altxt('Exercise Name')
    fireEvent.change(exerciseNameInputs[0], changeEvent('Random name'))

    fireEvent.click(t('change-state-btn'))
    fireEvent.click(qat('state-btn')[0])

    expect(qt('modal-wrapper')).toBeInTheDocument()
  })

  it('discard the changes if discard button on modal is clicked', () => {
    const {
      getAllByLabelText: altxt,
      getByTestId: t,
      queryAllByTestId: qat,
      queryByText: qtxt,
    } = component({
      template: filledTemplateTableWithData,
    })
    const exerciseNameInputs = altxt('Exercise Name')
    fireEvent.change(exerciseNameInputs[0], changeEvent('Random name'))

    fireEvent.click(t('change-state-btn'))
    fireEvent.click(qat('state-btn')[0])

    fireEvent.click(qtxt('Discard Changes'))
    filledTemplateTableWithData.exercises.forEach((e) => {
      // expect(t('exercise-list-wrapper').textContent).toContain(e.exerciseData.name)
      expect(
        altxt('Exercise Name').findIndex((el) => {
          return (el as HTMLInputElement).value === e.exerciseData.name
        })
      ).not.toBe(-1)
    })
  })

  it('saves the changes if save button on modal is clicked', () => {
    const handleSave = jest.fn()

    const {
      getAllByLabelText: altxt,
      getByTestId: t,
      queryAllByTestId: qat,
      queryByText: qtxt,
    } = component({
      template: filledTemplateTableWithData,
      handleSave,
    })
    const exerciseNameInputs = altxt('Exercise Name')
    const newVal = 'Random name'
    const oldVal = (exerciseNameInputs[0] as HTMLInputElement).value

    fireEvent.change(exerciseNameInputs[0], changeEvent(newVal))

    fireEvent.click(t('change-state-btn'))
    fireEvent.click(qat('state-btn')[0])

    fireEvent.click(qtxt('Save Changes'))

    const { exercises, ...rest } = filledTemplateTableWithData

    const changedIndex = exercises.findIndex((e) => e.exerciseData.name === oldVal)

    expect(handleSave).toBeCalled()
    expect(handleSave).toBeCalledWith(
      {
        ...rest,
        exercises: exercises
          .map((e) => {
            const {
              exerciseData: { created_by, ...exerciseData },
              ...rest
            } = e

            return {
              ...rest,
              exerciseData,
            }
          })
          .map((e, i) => {
            if (i !== changedIndex) {
              return e
            }

            const { exerciseData, ...rest } = e

            return {
              ...rest,
              exerciseData: {
                ...exerciseData,
                name: newVal.toUpperCase(),
                id: -1,
              },
            }
          }),
      },
      true
    )
  })

  it('renders a modal if delete button is clicked', () => {
    const { getByLabelText: l, getByTestId: t } = component()
    fireEvent.click(t('change-state-btn'))
    fireEvent.click(l('delete template'))

    expect(t('modal-wrapper')).toBeInTheDocument()
  })

  it('calls the provided handler function when delete button is clicked', () => {
    const handleDelete = jest.fn()
    const { getByLabelText: l, getByTestId: t, getByText: txt } = component({ handleDelete })
    fireEvent.click(t('change-state-btn'))
    fireEvent.click(l('delete template'))
    fireEvent.click(txt('Delete'))

    expect(handleDelete).toBeCalled()
    expect(handleDelete).toBeCalledWith(defaultTemplateTableWithData.id)
  })

  describe('Default creating behavior.', () => {
    const create = (args: Partial<Props> = {}) =>
      component({ defaultState: PageState.CREATE, ...args })

    it('renders the name input with an empty value and exercise list with empty children', () => {
      const { getByLabelText: l, queryAllByTestId: qat } = create()
      expect((l('TEMPLATE NAME') as HTMLInputElement).value).toBe('')
      expect(qat('exercise-list-item-wrapper').length).toBe(1) // just render the add button.
    })

    it('should disallow user to change the page state', () => {
      const { queryAllByTestId: qat, getByTestId: t } = create()
      fireEvent.click(t('change-state-btn'))
      expect(qat('state-btn').length).toBe(0)
    })
  })

  describe('View state', () => {
    it('hides the color picker', () => {
      const { queryByText: qtxt } = component({ defaultState: PageState.VIEW })
      expect(qtxt('CHANGE COLOR')).not.toBeInTheDocument()
    })
  })
})
