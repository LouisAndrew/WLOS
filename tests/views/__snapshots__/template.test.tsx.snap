// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template page matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div>
      <div
        class="wrapper"
        data-testid="template-wrapper"
      >
        <div
          class="header"
        >
          <button
            aria-describedby="popup-10"
            class="btn btn--xs btn--ghost page-state"
            data-testid="change-state-btn"
          >
            EDITING
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17.849 11.808l-.707-.707-9.9 9.9H3v-4.243L14.313 5.444l5.657 5.657a1 1 0 0 1 0 1.414l-7.07 7.071-1.415-1.414 6.364-6.364zm-2.121-2.121l-1.415-1.414L5 17.586v1.415h1.414l9.314-9.314zm2.828-7.071l2.829 2.828a1 1 0 0 1 0 1.414L19.97 8.273 15.728 4.03l1.414-1.414a1 1 0 0 1 1.414 0z"
                />
              </g>
            </svg>
          </button>
          <div
            class="name-wrapper"
          >
            <div
              aria-label="Template Color"
              class="color-indicator"
              data-testid="color-indicator"
              style="background-color: rgb(14, 14, 14);"
            />
            <label
              class="name-input-label"
            >
              TEMPLATE NAME
              <input
                placeholder="ENTER NAME"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="button-group"
          >
            <button
              aria-describedby="popup-14"
              class="btn btn--ghost btn--xs"
            >
              CHANGE COLOR
            </button>
          </div>
        </div>
        <div>
          <span
            class="exercise-text"
          >
            ADD EXERCISES TO TEMPLATE
          </span>
          <div
            data-testid="exercise-list-wrapper"
          >
            <div
              class="exercise-list"
              data-rbd-droppable-context-id="1"
              data-rbd-droppable-id="exercise-list"
            />
            <div
              class="wrapper buttons mt-2 duration-200"
              data-testid="exercise-list-item-wrapper"
              style="opacity: 1; transform: scale(1);"
            >
              <button
                class="btn btn--s btn--secondary btn--s"
                data-testid="new-exercise"
              >
                CREATE NEW EXERCISE
              </button>
              <button
                class="btn btn--s btn--primary btn--s list-button"
              >
                PICK FROM LIST
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-1"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-1-hidden-text-2"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
      data-testid="template-wrapper"
    >
      <div
        class="header"
      >
        <button
          aria-describedby="popup-10"
          class="btn btn--xs btn--ghost page-state"
          data-testid="change-state-btn"
        >
          EDITING
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M17.849 11.808l-.707-.707-9.9 9.9H3v-4.243L14.313 5.444l5.657 5.657a1 1 0 0 1 0 1.414l-7.07 7.071-1.415-1.414 6.364-6.364zm-2.121-2.121l-1.415-1.414L5 17.586v1.415h1.414l9.314-9.314zm2.828-7.071l2.829 2.828a1 1 0 0 1 0 1.414L19.97 8.273 15.728 4.03l1.414-1.414a1 1 0 0 1 1.414 0z"
              />
            </g>
          </svg>
        </button>
        <div
          class="name-wrapper"
        >
          <div
            aria-label="Template Color"
            class="color-indicator"
            data-testid="color-indicator"
            style="background-color: rgb(14, 14, 14);"
          />
          <label
            class="name-input-label"
          >
            TEMPLATE NAME
            <input
              placeholder="ENTER NAME"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="button-group"
        >
          <button
            aria-describedby="popup-14"
            class="btn btn--ghost btn--xs"
          >
            CHANGE COLOR
          </button>
        </div>
      </div>
      <div>
        <span
          class="exercise-text"
        >
          ADD EXERCISES TO TEMPLATE
        </span>
        <div
          data-testid="exercise-list-wrapper"
        >
          <div
            class="exercise-list"
            data-rbd-droppable-context-id="1"
            data-rbd-droppable-id="exercise-list"
          />
          <div
            class="wrapper buttons mt-2 duration-200"
            data-testid="exercise-list-item-wrapper"
            style="opacity: 1; transform: scale(1);"
          >
            <button
              class="btn btn--s btn--secondary btn--s"
              data-testid="new-exercise"
            >
              CREATE NEW EXERCISE
            </button>
            <button
              class="btn btn--s btn--primary btn--s list-button"
            >
              PICK FROM LIST
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
